lib/view_component/base.rb

```diff
@@ -328,7 +330,7 @@ def content_evaluated?
     end
 
     def maybe_escape_html(text)
-      return text if __vc_request && !__vc_request.format.html?
+      return text if @current_template && !@current_template.html?
```

Comment:
Note that current_template is set by `render_template_for` now, so this logic takes into account the format picked by the renderer (e.g. if the chosen template is `json` then the before/after strings won't be HTML sanitised).